objects/src/constants.rs

```diff
@@ -0,0 +1,63 @@
+/// Depth of the account database tree.
+pub const ACCOUNT_TREE_DEPTH: u8 = 64;
+
+/// The depth of the Merkle tree used to commit to notes produced in a block.
+pub const NOTE_TREE_DEPTH: u8 = 20;
+
+/// The maximum number of assets that can be stored in a single note.
+pub const MAX_ASSETS_PER_NOTE: usize = 256;
+
+/// The maximum number of inputs that can accompany a single note.
+///
+/// The value is set to 128 so that it can be represented using as a single byte while being
+/// evenly divisible by 8.
+pub const MAX_INPUTS_PER_NOTE: usize = 128;
+
+/// The maximum number of notes that can be consumed by a single transaction.
+pub const MAX_INPUT_NOTES_PER_TX: usize = 1023;
+
+/// The maximum number of new notes created by a single transaction.
+pub const MAX_OUTPUT_NOTES_PER_TX: usize = 4096;
+
+/// The minimum proof security level used by the Miden prover & verifier.
+pub const MIN_PROOF_SECURITY_LEVEL: u32 = 96;
+
+// TRANSACTION BATCH
+// ================================================================================================
+
+/// The depth of the Sparse Merkle Tree used to store created notes in a single batch.
+///
+/// A single note uses two leaves in the SMT. The even leaf is used to store the note's id, the odd
+/// leaf is used to store the note's metadata.
+pub const BATCH_CREATED_NOTES_SMT_DEPTH: u8 = 13;
```

Comment:
We are switching to the new terminology: "output note" instead of "created note" (not everything is renamed yet though). To be consistent with the new convention, this would be `BATCH_OUTPUT_NOTE_SMT_DEPTH`.

But also, we could probably name this `BATCH_NOTE_TREE_DEPTH`. I've also changed `SMT` to `TREE` as I think `SMT` part is an implementation detail.